### PR TITLE
EP+pad support for   Step-3.5-Flash-FP8

### PR DESCRIPTION
**Pull request by LJ-underdog**
## Motivation

Enable end-to-end serving of **stepfun Step-3.5-Flash-FP8** (FP8 block-quantized MoE) on ATOM at TP8 on AMD gfx942 (MI308X), across the two production parallelism paths: **Expert-Parallel (EP)** and **pure Tensor-Parallel with padding**.

## Technical Details

This branch adds the support required to load and run Step-3.5-Flash-FP8:

- **Step-3.5-Flash model support**: new `step3p5` model definition and MoE weight shuffling for the routed-expert layout.
- **FP8 block-quantized MoE inference**: per-`1x128` block scaling (`block_shape = [128, 128]`), with correct `inter_dim` padding-alignment across all TP configurations, plumbed through `Fp8MoEMethod.get_fused_moe_quant_config`.
- **TP8 FP8 weight loading**: handle `D < tp_size` in the fp8 `_load_w13` / `_load_w2` shard loaders so the routed-expert weights load correctly at TP8.
- **Sliding-window attention (SWA) per-layer KV-head workspace**: Step-3.5-Flash enables `sliding_window`. `ModelConfig.get_num_kv_heads()` returns the model-level value (32), but the per-layer attention modules use 4 KV heads; the SWA workspace is now sized from the true **per-layer** `num_kv_heads` (max across layers), so the workspace head dimension is correct. Touches only `atom/plugin/attention.py`.
- **Two production parallelism paths**:
  - **EP** (`--enable-expert-parallel`): experts sharded, `inter_dim = 1280`.
  - **Pure-TP + pad**: `inter_dim` sharded `1280/8 = 160`, padded to 256.

## Test Plan

End-to-end correctness via the ATOM-native `simple_inference` example on 8×gfx942 (TP8, FP8 block-quantized MoE, FP8 KV cache), over the example's 4 prompts, for both production paths:

- **EP**: `--enable-expert-parallel`.
- **Pure-TP + pad**: default (no expert-parallel).

## Test Result

Both paths **4/4 coherent** (exit 0), non-garbled, with natural EOS and no faults:

- **EP** (inter=1280): 4/4 prompts coherent; arithmetic correct (`1 + 2 + 3 = 6`).
- **Pure-TP + pad** (inter=256): 4/4 prompts coherent; Engine Core fully initialized (44/44 shards); arithmetic correct (`1 + 2 + 3 = 6`).
- No GPU faults; VRAM returned cleanly to baseline after each run (no leak).

## Submission Checklist

- [x] I have read and followed the [contributing guidelines](https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests).
- [x] This PR targets the default integration branch (`main`).
- [x] The code builds successfully.
- [x] I have included the log of a successful test run (EP + pure-TP-pad e2e, 4/4 coherent — logs available on request).
- [x] This PR does not break existing test cases.
- [x] Dependent changes are identified below (aiter production PR + CK pin).

## Related

- Depends on the **aiter production PR** ROCm/aiter#3548 (SwigluStep MoE activation + Composable Kernel pinned to upstream-clean `af7118e`). Note: the SwigluStep host codegen is not yet in aiter `main`, so this stack requires that aiter branch.
- ROCm/rocm-libraries#7920 — an orthogonal Composable Kernel correctness fix, upstreamed independently.
- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: atom/model_ops/moe.py [hunk 1]
```diff
@@ -1067,6 +1103,22 @@ def apply(
...
+                    if hasattr(self, "moe")
+                    else getattr(self, "ep_size", 1)
+                )
+                if layer.reduce_results and (_tp_size > 1 or _ep_size > 1):
+                    from aiter.dist.parallel_state import get_tp_group
+
+                    _moe_result = get_tp_group().all_reduce(
+                        _moe_result, ca_fp8_quant=False
+                    )
                 return _moe_result
```
[the remark this answers is not in the document]
Comment (LJ-underdog) [reply atom/model_ops/moe.py]: Fixed in 94e031d4: removed the redundant all_reduce in Mxfp4MoEMethod.apply — other apply() methods don't reduce internally, forward_impl already all-reduces when reduce_results is set, so reduction now happens once. Good catch

File: atom/model_ops/moe.py [hunk 2]
```diff
@@ -2623,12 +2749,29 @@ def _load_w2(
...
+                if expert_data.dtype == torch.float32:
+                    if load_shard_size != shard_size:
+                        expert_data.narrow(shard_dim, 0, load_shard_size).zero_()
+                    else:
+                        expert_data.zero_()
+                return
+            size = min(load_shard_size, loaded_weight.shape[shard_dim] - start)
+            loaded_weight = loaded_weight.narrow(shard_dim, start, size)
+            if load_shard_size != shard_size:
+                expert_data = expert_data.narrow(shard_dim, 0, load_shard_size)
```
[the remark this answers is not in the document]
Comment (LJ-underdog) [reply atom/model_ops/moe.py]: Fixed in 5663a664: expert_data now narrowed to actual shard size with remainder zeroed — trailing partial rank (e.g. tp=4, D=10) no longer hits copy_() shape mismatch. No change for tp=8

File: atom/model_ops/attentions/aiter_attention.py [hunk 3]
```diff
@@ -59,7 +59,15 @@ def __init__(
...
-        num_head_k = max(1, hf_config.num_key_value_heads // get_tp_group().world_size)
+        num_head_k = max(
+            1,
+            getattr(
+                hf_config,
+                "num_key_value_heads",
+                getattr(hf_config, "num_attention_groups", None),
+            )
+            // get_tp_group().world_size,
+        )
```
[the remark this answers is not in the document]
Comment (LJ-underdog) [reply atom/model_ops/attentions/aiter_attention.py]: Fixed in 4ea9985f: raises a clear ValueError when neither num_key_value_heads nor num_attention_groups is present (mirrors model_runner), instead of a None // world_size TypeError.

File: atom/models/step3p5.py [hunk 4]
```diff
@@ -0,0 +1,890 @@
...
+        sliding_window = None
+        sinks = None
+        if is_sliding:
+            sliding_window = getattr(config, "sliding_window", None)
+            # config.sink may be False (bool) or an int; only a positive int
+            # enables attention sinks (False/0 -> no sinks).
+            sink_size = getattr(config, "sink", 0)
+            if sink_size > 0:
+                sinks = nn.Parameter(torch.empty(self.num_heads, requires_grad=False))
+
```
[the remark this answers is not in the document]
Comment (LJ-underdog) [reply atom/models/step3p5.py:447]: Fixed in 4ea9985f: sinks now zero-initialized (torch.empty →torch.zeros)
